### PR TITLE
Introduce build process of the documentation using slatedocs and github actions

### PR DESCRIPTION
**Pull request by sprilukintibco**
# Description
- after this change all document updates should be done inside `./slatedocs` folder
- once changes are pushed to the master - github actions automatically build and publush documentation to github pages (gh-pages branch)
- local development can be done with the help of the docker see [slatedocs](https://github.com/slatedocs/slate/wiki/Using-Slate-in-Docker) documentation

## Live document preview
Document preview for this PR is available [here](https://sprilukintibco.github.io/js-visualize)

## To Review: 
- Documentation header has been changed: need to get approval for it:
Screen Shot 2021-02-07 at 00 26 03

- Readme.md in the root of the master branch has been updated to not contain visualize documentation, instead it contains description for developers about how to update viz documentation: 
  - how to add/update samples)
  - how to update documentation itself
  - how to develop documentation locally with the help of slatedocs dev server

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,628 +1,190 @@
...
+
+For other ways to build documentation - see slate [getting started][slatedocs getting started] docs
+
+### Update slatedocs template to the latest version
+Template is slightly changed but generally it should be easy to update slate template to the latest version
+if necessary. To update template:
+- clone original [slatedocs repo][slatedocs]
+- copy all files and folders from the cloned repo except `.github` and `source/index.html.md` to `slatedocs` 
+  folder and replace files if they exists
+- copy `.github` folder from cloned the repo to the root `.github` folder and replace files if they exists.
```
Comment (yplakosh-tibco) [inline Readme.md:120]: Change "folder from cloned the repo to the root" to "folder from the cloned repo to the root"

File: Readme.md [hunk 2]
```diff
@@ -1,628 +1,190 @@
...
+- use some git diff tool to inspect difference and merge changes.
+
+## Github pages
+[Github pages][github pages] used to made documentation site available online.
```
Comment (yplakosh-tibco) [inline Readme.md:124]: Change "used to made" to "used to make"

File: Readme.md [hunk 3]
```diff
@@ -1,628 +1,190 @@
...
+To update documentation, please fork this repo and set up github pages to be server from a `gh-pages` branch:
```
Comment (yplakosh-tibco) [inline Readme.md:125]: Change "to be server from" to "to be served from"

File: Readme.md [hunk 4]
```diff
@@ -1,628 +1,190 @@
...
+- Then select `gh-pages` branch as a source and click `save`
+- After this your site should be available at this url: `https://<username>.github.io/js-visualize/`
+
+## Github actions
+[Github actions][github actions] are used to automatically build and deploy the public site.
+Two actions: `Build` and `Deploy` are created:
+- `Build` action runs on push/pull request to any branch and makes sure that documentation
+  build process executed successfully
+- `Deploy` action runs on push to the `master` branch.
+   it executes documentation build process and then
```
Comment (yplakosh-tibco) [inline Readme.md:136]: Change "it" to "It"

File: Readme.md [hunk 5]
```diff
@@ -1,628 +1,190 @@
...
+   it executes documentation build process and then 
+   it executes push of the `slatedocs/buld` folder to the `gh-pages` branch which
+   updates the public documentation if [Github pages](#github-pages) has been set up.
+
+## How to update documentation
+- Fork this repo
+- Set up [Github pages](#github-pages) for your fork  
+- If you already forked the repo earlier - [reset](#how-to-reset-your-fork-to-mirror-original-repo) 
+  your fork's `master` branch to mirror the original repo's `master` branch  
+- Made necessary changes to the `slatedocs/source/index.html.md` file in your fork
```
Comment (yplakosh-tibco) [inline Readme.md:145]: Change "Made necessary " to "Make necessary "

**Review by yplakosh-tibco — COMMENTED** [review]
@sprilukintibco please add a comment with a notice to all slate files that were modified compared to the original code. The notice should state that the file has been modified with a short description of the modification/s.

File: Readme.md [hunk 6]
```diff
@@ -1,628 +1,190 @@
...
+- [Github pages](#github-pages)
+- [Github actions](#github-actions)
+- [How to update documentation](#how-to-update-documentation)
+  - [How to reset your fork to mirror original repo](#how-to-reset-your-fork-to-mirror-original-repo)
+- [License](#license)  
+
+## Project structure
+Visualize.js documentation project consists of 3 parts:
+- **jsfiddle examples** - this is basically set of folders with jsfiddle examples. 
+  we are using [jsfiddle github integration][jsfiddle github integration]
```
[the remark this answers is not in the document]
Comment (sprilukintibco) [reply Readme.md]: fixed

File: Readme.md [hunk 7]
```diff
@@ -1,628 +1,190 @@
...
+- **slatedocs** - [visualize.js docs page][online documentation] 
+  is written using [slatedocs/slate][slatedocs] template. So if visualize document itself
+  should be updated - read [this](#slatedocs) section
+- **github pages** - final result of the documentation made public using [github pages][github pages],
+    namely `gh-pages` branch is used to show public documentation site
+
+## Folders structure
+This github project has the following structure:
 ```
+|-<ahv-*> - adhoc view samples
```
[the remark this answers is not in the document]
Comment (sprilukintibco) [reply Readme.md]: fixed

File: Readme.md (hunk 1 above)
Comment (sprilukintibco) [reply Readme.md]: fixed

File: Readme.md (hunk 2 above)
Comment (sprilukintibco) [reply Readme.md]: fixed

File: Readme.md (hunk 3 above)
Comment (sprilukintibco) [reply Readme.md]: fixed

File: Readme.md (hunk 4 above)
Comment (sprilukintibco) [reply Readme.md]: fixed

File: Readme.md (hunk 5 above)
Comment (sprilukintibco) [reply Readme.md]: fixed

**Comment by sprilukintibco** [comment]
> @sprilukintibco please add a comment with a notice to all slate files that were modified compared to the original code. The notice should state that the file has been modified with a short description of the modification/s.

fixed

**Pull request merged**